ignite/metrics/epoch_metric.py

```diff
@@ -133,9 +136,15 @@ def compute(self) -> float:
             # Run compute_fn on zero rank only
             result = self.compute_fn(_prediction_tensor, _target_tensor)
 
+        # compute_fn outputs: scalars, tensors, tuple/list/mapping of tensors.
+        if not is_scalar_or_collection_of_tensor(result):
```

Comment:
I think there is no need to introduce a public `is_scalar_or_collection_of_tensor` for that. Let's create a small private function here.